src/azure-cli/azure/cli/command_modules/network/_params.py

```diff
@@ -146,6 +146,13 @@ def load_arguments(self, _):
         c.argument('enable_http2', arg_type=get_three_state_flag(positive_label='Enabled', negative_label='Disabled'), options_list=['--http2'], help='Use HTTP2 for the application gateway.', min_api='2017-10-01')
         c.ignore('public_ip_address_type', 'frontend_type', 'subnet_type')
 
+    with self.argument_context('network application-gateway', arg_group='Private Link Configuration') as c:
+        c.argument('enable_private_link', action='store_true', help='Enable Private Link feature for this application gateway', default=False)
```

Comment:
should user provide `frontend_ip` as well?

Reply:
The creation of appgw while specifying frontend IP can take effect only one IP, either the private one or the public one. 
No matter which one is selected, it's named `appGatewayFrontendIP`. 
So, requiring user to specify a --frontend-ip here is unnecessary.
Besides, I left the help message `The frontend IP which the Private Link will associate to` in `--frontend-ip` to indicate user while adding a private link.